reservation_units/tasks.py

```diff
@@ -26,6 +29,80 @@ def _update_reservation_unit_pricings() -> None:
     logger.info(f"Updated {num_updated} reservation units with date {today}")
 
 
+@app.task(name="update_reservation_unit_pricings_tax_percentage")
+def update_reservation_unit_pricings_tax_percentage(
+    change_date: str,
+    current_tax: str,
+    future_tax: str,
+    ignored_company_codes: list[str] | tuple[str] = (),
+) -> None:
+    from reservation_units.models import ReservationUnitPricing, TaxPercentage
+
+    SentryLogger.log_message(
+        message="Task `update_reservation_unit_pricings_tax_percentage` started",
+        details=f"Task was run with "
+        f"change_date: {change_date}, "
+        f"current_tax: {current_tax}, "
+        f"future_tax: {future_tax}, "
+        f"ignored_company_codes: {ignored_company_codes}",
+        level="info",
+    )
+
+    change_date = date.fromisoformat(change_date)  # e.g. "2024-09-01"
+    current_tax, _ = TaxPercentage.objects.get_or_create(value=Decimal(str(current_tax)))
+    future_tax, _ = TaxPercentage.objects.get_or_create(value=Decimal(str(future_tax)))
+
+    # Last pricing for each reservation unit before the change date
+    latest_pricings = (
+        ReservationUnitPricing.objects.filter(
+            Q(begins__lte=change_date, pricing_type=PricingType.FREE)  # Ignore FREE pricings after the change date
+            | Q(pricing_type=PricingType.PAID)
+        )
+        .filter(status__in=(PricingStatus.PRICING_STATUS_ACTIVE, PricingStatus.PRICING_STATUS_FUTURE))
+        .exclude(reservation_unit__payment_accounting__company_code__in=ignored_company_codes)
+        .order_by("reservation_unit_id", "-begins")
+        .distinct("reservation_unit_id")
+    )
+    for pricing in latest_pricings:
+        # Skip pricings that are FREE or have a different tax percentage
+        # We don't want to filter these away in the queryset, as that might cause us to incorrectly create new pricings
+        # in some cases. e.g. Current pricing is PAID, but the future pricing is FREE or has a different tax percentage.
+        if (
+            pricing.pricing_type == PricingType.PAID
+            and pricing.highest_price > 0
+            and pricing.tax_percentage == current_tax
+            # Don't create a new pricing if the reservation unit has a future pricing after the change date
+            and pricing.begins < change_date
+        ):
+            pricing.id = None  # Create a new pricing when saving
```

Comment:
Is this documented behavior? It's a neat trick at least!

Reply:
Not sure if it's documented, but essentially it's the same as creating a copy of the object, but without the `id` field 😄